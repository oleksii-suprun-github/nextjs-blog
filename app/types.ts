export type SanityCategory = {
  title?: string;
  description?: string;
};

export type SanityPostMeta = {
  title: string;
};

type SanityPostBase = {
  title: string;
  category: {
    title: string;
    url: string;
  };
  slug: string;
  image: {
    url: string;
    alt?: string;
  };
  createdAt: string;
};

export type SanityPostPreview = SanityPostBase & {
  previewText: string;
};

export type SanityPost = SanityPostBase & {
  content: string;
};
