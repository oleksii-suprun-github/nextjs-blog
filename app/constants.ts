export const SITE_BRAND_TITLE_ENDING = '| NextJS Sample Blog';

export const NEXTJS_URL = '/publications/nextjs';

export const NODEJS_URL = '/publications/nodejs';

export const PYTHON_URL = '/publications/python';
