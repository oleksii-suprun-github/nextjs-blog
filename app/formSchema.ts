import { z } from 'zod';

export const formSchema = z.object({
  name: z
    .string()
    .min(2, { message: 'Name must be at least 2 characters long' })
    .max(50, { message: 'Name must be less than 50 characters' }),
  email: z.string().email({ message: 'Please enter a valid email address' }),
  message: z
    .string()
    .min(10, { message: 'Message must be at least 10 characters long' })
    .max(500, { message: 'Message must be less than 500 characters' }),
});
