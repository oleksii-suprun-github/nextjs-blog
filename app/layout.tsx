import { SpeedInsights } from '@vercel/speed-insights/next';
import './globals.css';

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html className="scroll-smooth" lang="en">
      <body className={'min-h-dvh text-stone-300'}>
        {children}
        <SpeedInsights />
      </body>
    </html>
  );
}
