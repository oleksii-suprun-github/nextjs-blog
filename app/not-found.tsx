import NotFound from './components/NotFound';

export const metadata = {
  title: 'Page Not Found | Agora Energiewende',
  description: 'Sorry, we can`t find that publication.',
};

const PageNotFound = () => <NotFound link="/publications" />;

export default PageNotFound;
