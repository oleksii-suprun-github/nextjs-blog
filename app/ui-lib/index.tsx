export { default as Loading } from './Loading';
export { default as HeadlineWithDescription } from './HeadlineWithDescription';
export { default as PublicationsSection } from './PublicationsSection';
export { default as PublicationCard } from './PublicationCard';
export { default as PostHero } from './PostHero';
export { default as Form } from './Form';
export { default as FormInput } from './FormInput';
export { default as Modal } from './Modal';
export { default as Drawer } from './Drawer';
