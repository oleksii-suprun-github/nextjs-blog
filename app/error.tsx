'use client';

import { useEffect } from 'react';
import ErrorMessage from './components/ErrorMessage';
import { SITE_BRAND_TITLE_ENDING } from './constants';

export const metadata = {
  title: `Something went wrong ${SITE_BRAND_TITLE_ENDING}`,
};

export default function GlobalError({
  error,
  reset,
}: {
  error: Error & { digest?: string };
  reset: () => void;
}) {
  useEffect(() => {
    console.error('An unexpected error occurred:', error);
  }, [error]);

  return (
    <html>
      <body>
        <ErrorMessage error={error} reset={reset} />
      </body>
    </html>
  );
}
