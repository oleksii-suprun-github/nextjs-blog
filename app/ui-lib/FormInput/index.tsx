import { FeedbackFormData } from '@/app/types';
import { FieldErrors, UseFormRegisterReturn } from 'react-hook-form';

type FormInputType = {
  errors: FieldErrors<FeedbackFormData>;
  type: string;
  register: UseFormRegisterReturn<string>;
  placeholder: string;
};

function FormInput({ errors, type, register, placeholder }: FormInputType) {
  return (
    <div className="mb-5 flex flex-col">
      {type === 'textarea' ? (
        <textarea
          {...register}
          className="textarea textarea-bordered resize-none bg-purple-950 p-4 placeholder:text-gray-400"
          placeholder={placeholder}
          rows={5}
        ></textarea>
      ) : (
        <input
          type={type}
          {...register}
          className="input input-bordered bg-purple-950 placeholder:text-gray-400"
          placeholder={placeholder}
        />
      )}
      {errors.name && <small className="mt-2 text-red-700">{errors.name.message}</small>}
    </div>
  );
}

export default FormInput;
