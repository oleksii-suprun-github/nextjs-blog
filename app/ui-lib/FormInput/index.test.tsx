import { render, screen } from '@testing-library/react';
import FormInput from '.';
import { FieldErrorsImpl } from 'react-hook-form';
import { FeedbackFormData } from '@/app/types';

const mockRegister = {
  name: 'test',
  onBlur: vi.fn(),
  onChange: vi.fn(),
  ref: vi.fn(),
};

const mockErrors: FieldErrorsImpl<FeedbackFormData> = {
  name: { type: 'required', message: 'Name is required' },
};

describe('FormInput Component', () => {
  test('renders input element correctly', () => {
    render(<FormInput errors={{}} type="text" register={mockRegister} placeholder="Your Name" />);

    const inputElement = screen.getByPlaceholderText('Your Name');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'text');
  });

  test('renders textarea element correctly', () => {
    render(
      <FormInput errors={{}} type="textarea" register={mockRegister} placeholder="Your Feedback" />,
    );

    const textareaElement = screen.getByPlaceholderText('Your Feedback');
    expect(textareaElement).toBeInTheDocument();
    expect(textareaElement).toHaveAttribute('rows', '5');
  });

  test('displays error message when error is present', () => {
    render(
      <FormInput errors={mockErrors} type="text" register={mockRegister} placeholder="Your Name" />,
    );

    const errorMessage = screen.getByText('Name is required');
    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveClass('text-red-700');
  });

  test('does not display error message when no error is present', () => {
    render(<FormInput errors={{}} type="text" register={mockRegister} placeholder="Your Name" />);

    const errorMessage = screen.queryByText('Name is required');
    expect(errorMessage).not.toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <FormInput errors={{}} type="text" register={mockRegister} placeholder="Your Name" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
