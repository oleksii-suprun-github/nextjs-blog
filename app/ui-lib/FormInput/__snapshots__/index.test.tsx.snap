// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormInput Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-5 flex flex-col"
  >
    <input
      class="input input-bordered bg-purple-950 placeholder:text-gray-400"
      name="test"
      placeholder="Your Name"
      type="text"
    />
  </div>
</DocumentFragment>
`;
